assets/js/src/segments/dynamic/dynamic_segments_filters/subscriber_subscribed_to_list.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect } from 'react';
+import { compose, get, filter } from 'lodash/fp';
+import { useDispatch, useSelect } from '@wordpress/data';
+
+import MailPoet from 'mailpoet';
+import Select from 'common/form/select/select';
+import ReactSelect from 'common/form/react_select/react_select';
+
+import {
+  AnyValueTypes,
+  SelectOption,
+  StaticSegment,
+  WordpressRoleFormItem,
+} from '../types';
+
+export function validateSubscribedToList(formItems: WordpressRoleFormItem): boolean {
+  return (
+    (
+      (formItems.operator === AnyValueTypes.ANY)
+      || (formItems.operator === AnyValueTypes.ALL)
+      || (formItems.operator === AnyValueTypes.NONE)
+    )
+    && (
+      (Array.isArray(formItems.segments))
+      && (formItems.segments.length > 0)
+    )
+  );
+}
+
+type Props = {
+  filterIndex: number;
+}
+
+export const SubscribedToList: React.FunctionComponent<Props> = ({ filterIndex }) => {
+  const segment: WordpressRoleFormItem = useSelect(
+    (select) => select('mailpoet-dynamic-segments-form').getSegmentFilter(filterIndex),
+    [filterIndex]
+  );
+  const staticSegmentsList: StaticSegment[] = useSelect(
+    (select) => select('mailpoet-dynamic-segments-form').getStaticSegmentsList(),
+    []
+  );
+
+  const { updateSegmentFilter, updateSegmentFilterFromEvent } = useDispatch('mailpoet-dynamic-segments-form');
+
+  useEffect(() => {
+    if (
+      (
+        (segment.operator !== AnyValueTypes.ANY)
+        && (segment.operator !== AnyValueTypes.ALL)
+        && (segment.operator !== AnyValueTypes.NONE)
+      )
+    ) {
+      updateSegmentFilter({ operator: AnyValueTypes.ANY }, filterIndex);
+    }
+  }, [updateSegmentFilter, segment, filterIndex]);
+  const options = staticSegmentsList.map((currentValue) => ({
+    value: currentValue.id.toString(),
+    label: currentValue.name,
+  }));
+
+  return (
+    <>
+      <Select
+        key="select"
+        value={segment.operator}
+        onChange={(e) => {
+          updateSegmentFilterFromEvent('operator', filterIndex, e);
+        }}
+      >
+        <option value={AnyValueTypes.ANY}>{MailPoet.I18n.t('anyOf')}</option>
+        <option value={AnyValueTypes.ALL}>{MailPoet.I18n.t('allOf')}</option>
+        <option value={AnyValueTypes.NONE}>{MailPoet.I18n.t('noneOf')}</option>
+      </Select>
+      <ReactSelect
+        dimension="small"
+        isFullWidth
+        isMulti
+        placeholder={MailPoet.I18n.t('searchLists')}
+        options={options}
+        value={
+          filter(
+            (option) => {
+              if (!segment.segments) return undefined;
+              const segmentId = Number(option.value);
```

Comment:
You shouldn't convert the value to `Number` here, because you'll compare it to an array of strings on the next line. Currently it will always return false, so pre-selected segments will be always empty.

Reply:
You are right, my thinking was to make everything nubmers because IDs of segments are stored in the database as strings. So I was thinking that we could maybe rather re-type the values to numbers in the server. What do you think?

Reply:
Strings or numbers, it's up to you, really, it just needs to be consistent and backward-compatible. In the [subscriber score segment PR](https://github.com/mailpoet/mailpoet/pull/3821) I decided to store values as strings, so if you change to numbers, be sure not to break it.